compute/accelerator/benchmarks/mamf-finder.py

```diff
@@ -233,15 +233,26 @@ def func_wrapper(*args, **kwargs):
         B = B.to(torch.float8_e4m3fn)
 
         # some torch versions require the scale arg, some don't so discover which is required
+        test_result = torch._scaled_mm(A, B)
```

Comment:
I don't follow you. You were the one that proposed to add the version with scale:
https://github.com/stas00/ml-engineering/pull/85

You said:
> The second issue is that for certain versions of PyTorch, torch._scaled_mm requires specifying scale_a and scale_b, otherwise, an error will be thrown due to the absence of default values.

So that proposed code will fail on some pt versions since there is no scale args in it.

So what you're saying is that now there is yet another version of `torch._scaled_mm` that returns a tuple, right?

I'm asking the torch folks of the right way to proceed.

Reply:
I suppose the main cause of the problem is that the API isn't stable :( might have to require a specific pt version for fp8 - that will simplify the code a lot - perhaps 2.6 as it's about to be released?

Reply:
Previously, there was an issue with a specific version of `torch._scale_mm` where the input parameters required a scale. Now, for PyTorch 2.4, it not only returns the output but also returns a scale within a tuple. However, for PyTorch 2.5, it only returns a tensor computed with FP8.

Reply:
Yes, it's very unstable and apparently it isn't finished yet - will continue breaking/changing!

I should have the back-compatibility recipe soon. If you're on pytorch slack you can follow the discussion with the dev who is in charge of this new API here:
https://pytorch.slack.com/archives/C3PDTEV8E/p1736397352322799

Reply:
still waiting for the author to write it

Reply:
OK, here it is:
https://gist.github.com/drisspg/783616821043ab4594b9784f556c6714

Reply:
based on this clear guide I propose a simple solution here - let's require torch>=2.5 only when fp8 is wanted and then we don't have to support all those earlier versions - thoughts?

Reply:
Alright, I agree.

Reply:
but why did you close this PR? we could have adapted it here - or do you want to open a new PR? or should I do it? whatever you prefer

Reply:
Apologies, I mistakenly thought you had already adjusted this logic, and I can reopen it to make the modifications.